otx/algorithms/classification/tasks/inference.py

```diff
@@ -260,31 +259,28 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, update_progre
                 dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
 
             if saliency_map is not None:
-                actmap = get_actmap(saliency_map, (dataset_item.width, dataset_item.height))
-                saliency_map_media = ResultMediaEntity(
-                    name="Saliency Map",
-                    type="saliency_map",
-                    annotation_scene=dataset_item.annotation_scene,
-                    numpy=actmap,
-                    roi=dataset_item.roi,
+                if saliency_map.ndim == 4 and saliency_map.shape[0] == 1:
+                    saliency_map = saliency_map[0]
+                add_saliency_maps_to_dataset_item(
+                    dataset_item=dataset_item,
+                    saliency_map=saliency_map,
+                    model=self._task_environment.model,
+                    labels=self._labels,
```

Comment:
Can we move the if block to add_saliency_maps_to_dataset_item()?
Is the condition is common or task specific?

Reply:
There were some uncertainty with the input shape, I'll remove this line